taskvine/src/worker/vine_cache.c

```diff
@@ -422,7 +407,9 @@ int vine_cache_remove(struct vine_cache *c, const char *cachename, struct link *
 		return 0;
 
 	/* Ensure that any child process associated with the entry is stopped. */
-	vine_cache_kill(c, f, cachename, manager);
```

Comment:
I think that you want to still call `vine_cache_kill` even if `manager` is null, since that does other work.  Just check for manager null at the point where it actually gets used (sending a cache-update/invalid message, I think).